File: Ejercicio 014/electrodomestico.py
class Electrodomestico:
    def __init__(self, marca, consumo, color, estado):
        self.marca = marca
        self.consumo = consumo
        self.color = color
        self.estado = estado

    def mostrar_info(self):
        print("Marca:", self.marca)
        print("Consumo:", self.consumo, "W")
        print("Color:", self.color)
        print("Estado:", self.estado)

    def funcion(self):
        pass


class Nevera(Electrodomestico):
    def __init__(self, marca, consumo, color, estado, capacidad):
        super().__init__(marca, consumo, color, estado)
        self.capacidad = capacidad

    def funcion(self):
        print("Enfriando alimentos - Capacidad:", self.capacidad, "L")


class Lavadora(Electrodomestico):
    def __init__(self, marca, consumo, color, estado, carga):
        super().__init__(marca, consumo, color, estado)
        self.carga = carga

    def funcion(self):
        print("Lavando ropa - Carga:", self.carga, "kg")


class Microondas(Electrodomestico):
    def __init__(self, marca, consumo, color, estado, potencia):
        super().__init__(marca, consumo, color, estado)
        self.potencia = potencia

    def funcion(self):
        print("Calentando comida - Potencia:", self.potencia, "W")


e1 = Nevera("LG", 300, "Plateado", "Encendido", 400)
e2 = Lavadora("Samsung", 500, "Blanco", "Apagado", 12)
e3 = Microondas("Whirlpool", 1200, "Negro", "Encendido", 900)

lista = [e1, e2, e3]

for e in lista:
    print("Tipo:", type(e).__name__)
    e.mostrar_info()
    e.funcion()
    print("-------------------")

File: Ejercicio 014/electrodomestico.java
class Electrodomestico {
    String marca;
    int consumo;
    String color;
    String estado;

    public Electrodomestico(String marca, int consumo, String color, String estado) {
        this.marca = marca;
        this.consumo = consumo;
        this.color = color;
        this.estado = estado;
    }

    public void mostrarInfo() {
        System.out.println("Marca: " + marca);
        System.out.println("Consumo: " + consumo + "W");
        System.out.println("Color: " + color);
        System.out.println("Estado: " + estado);
    }

    public void funcion() {}
}

 
class Nevera extends Electrodomestico {
    int capacidad;

    public Nevera(String marca, int consumo, String color, String estado, int capacidad) {
        super(marca, consumo, color, estado);
        this.capacidad = capacidad;
    }

    public void funcion() {
        System.out.println("Enfriando alimentos - Capacidad: " + capacidad + "L");
    }
}


class Lavadora extends Electrodomestico {
    int carga;

    public Lavadora(String marca, int consumo, String color, String estado, int carga) {
        super(marca, consumo, color, estado);
        this.carga = carga;
    }

    public void funcion() {
        System.out.println("Lavando ropa - Carga: " + carga + "kg");
    }
}


class Microondas extends Electrodomestico {
    int potencia;

    public Microondas(String marca, int consumo, String color, String estado, int potencia) {
        super(marca, consumo, color, estado);
        this.potencia = potencia;
    }

    public void funcion() {
        System.out.println("Calentando comida - Potencia: " + potencia + "W");
    }
}

public class Main {
    public static void main(String[] args) {

        Electrodomestico e1 = new Nevera("LG", 300, "Plateado", "Encendido", 400);
        Electrodomestico e2 = new Lavadora("Samsung", 500, "Blanco", "Apagado", 12);
        Electrodomestico e3 = new Microondas("Whirlpool", 1200, "Negro", "Encendido", 900);

        Electrodomestico[] lista = {e1, e2, e3};

        for (Electrodomestico e : lista) {
            System.out.println("Tipo: " + e.getClass().getSimpleName());
            e.mostrarInfo();
            e.funcion();
            System.out.println("-------------------");
        }
    }
}
